packages/clerk-js/src/ui/components/UserProfile/UsernamePage.tsx

```diff
@@ -32,15 +44,28 @@ export const UsernamePage = withCardStateProvider(() => {
         headerTitle={localizationKeys('userProfile.usernamePage.title')}
         Breadcrumbs={UserProfileBreadcrumbs}
       >
+        {!canEdit && <InformationBox message={localizationKeys('userProfile.usernamePage.readonly')} />}
+
         <Form.Root onSubmit={updatePassword}>
           <Form.ControlRow elementId={usernameField.id}>
             <Form.Control
               {...usernameField.props}
               required
               autoFocus
+              isDisabled={!canEdit}
             />
           </Form.ControlRow>
-          <FormButtons isDisabled={!canSubmit} />
+          {canEdit ? (
+            <FormButtons isDisabled={!canSubmit} />
+          ) : (
+            <FormButtonContainer>
+              <Form.ResetButton
+                localizationKey={localizationKeys('userProfile.formButtonReset')}
+                block={false}
+                onClick={navigateToFlowStart}
```

Comment:
@nikosdouvlis Should we avoid further usage of this? wdyt?

Reply:
@nikosdouvlis any input here?

Reply:
Can we just use `navigate('..')` here?

Reply:
Note: I'm using the same navigation as per `<FormButtons />`.